### DISCUSSION TITLE
Dealing with time limited image links such as AWS S3 Images

### DISCUSSION THREAD

**Discussion by StephenBanz** [onevcat/Kingfisher#1915] [Q&A]
We were using redirection for authentication.
We had a bit of a problem with redirecting images and we decided to remove the extra and step and use the S3 link directly.

Everything else is working fine but since it has a time limit there have been some problems.
consider the following scenario,
- There are some images somewhere in the middle of the page 
- Open the page(S3 image links are received from API)
- Don't scroll to the image, just stay at the top of the page till the time limit is reached
- After the time limit scroll to the images

Now the images will be inaccessible. 

My wording might be wrong but I want to stop this lazy loading and implement eager loading in such scenario.
Is this possible?

**Comment by onevcat** [comment]
Maybe you can try the `ImagePrefetcher` (https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet#prefetch), which seems meets your needs perfectly.

When loading your page, give the prefetcher all your URLs and it would start the downloading. You just use the normal image view's extension methods (`imageView.kf.setImage(with: url)`) to set image of an URL to the image view. Kingfisher will try to find it in cache first. If it matches, the image is just set; if not, the image view will wait for `ImagePrefetcher` to finish the downloading and then set it.

Please read the related wiki in the link above for detail.
